config/common.php

```diff
@@ -0,0 +1,25 @@
+<?php
+declare(strict_types=1);
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\EventDispatcher\ListenerProviderInterface;
+use Yiisoft\EventDispatcher\Dispatcher;
+use Yiisoft\EventDispatcher\Provider\Provider;
+use Yiisoft\Factory\Definitions\Reference;
+use Yiisoft\Widget\Widget;
+use Yiisoft\Widget\Factory\WidgetFactory;
+
+return [
+    ListenerProviderInterface::class => [
+        '__class' => Provider::class,
+    ],
+
+    EventDispatcherInterface::class => [
```

Comment:
I phink for this case it better to use a custom Closure like:
```php
EventDispatcherInterface::class => fn (ContainerIntarface $c) => new Dispatcher($c->get(ListenerProviderInterface::class)),
```
because `Reference::to` already will create a Closure and this also does not exclude the autowiring overhead

Reply:
Done.